### ISSUE TITLE
InvalidRegistration when sending to topics

### ISSUE THREAD

**Issue by sapostolou** [ToothlessGear/node-gcm#278]
I'm trying to send a notification to a topic but I get the InvalidRegistration error.

This is the message's code:
```
var fcmMesssage = new gcm.Message({
		collapseKey: 'key1',
		priority: 'high',
		contentAvailable: true,
		notification: { 
			title: 'My app',
			message: 'New stuff',
			icon: "myicon",
			tag: 'mytag'
		}
	});
```

and the sender's code:
```
sender.send(fcmMesssage, { condition: '\'mytopic\' in topics'}, function (err, response) {  
		if (err){
			console.log(err);
		}
		else{
			console.log(response);
		}
	});
```

I had no problem sending the following message through the HTTP API:
```
{
    "condition":"'mytopic' in topics",
    "collapse_key":"demo",
    "priority": "high",
    "content_available":true,
    "notification":{
        "title":"My app",
        "body":"New stuff",
        "tag":"demo2"
    }
}
```

**Comment by taroutatsou** [comment]
I replaced `condition: '\'mytopic\' in topics'` with `condition: '\'mytopic\' in topics || \'mytopic2\' in topics'` and it worked.

I did that because of [this line ](https://github.com/ToothlessGear/node-gcm/blob/master/lib/sender.js#L201)in sender.js.

It will work for me anyway since I will always be using an actual condition but I think that it should be able to work without an actual logical AND or OR.

**Comment by eladnava** [comment]
@taroutatsou Thanks for reporting this!

@hypesystem I think we should modify `extractRecipient` to pass the type of recipient extracted to the callback and not just the recipient value [here](https://github.com/ToothlessGear/node-gcm/blob/master/lib/sender.js#L226).

What do you think?

**Comment by hypesystem** [comment]
The alternative is fixing the regex (which seems like a smaller change), but I don't really mind -- either solution is good :smile:

**Comment by eladnava** [comment]
To be honest I don't think that regex is a robust enough solution to begin with 🤕

**Comment by eladnava** [comment]
@taroutatsou Would you like to submit a PR? 😄

**Comment by eladnava** [comment]
Fixed via #312.
